packages/venia-ui/lib/components/OrderHistoryPage/OrderDetails/item.js

```diff
@@ -0,0 +1,99 @@
+import React, { useMemo } from 'react';
+import { shape, string, number, arrayOf } from 'prop-types';
+
+import { Link, resourceUrl } from '@magento/venia-drivers';
+import { mergeClasses } from '@magento/venia-ui/lib/classify';
+
+import Button from '../../Button';
+import ProductOptions from '../../LegacyMiniCart/productOptions';
+import Image from '../../Image';
+
+import defaultClasses from './item.css';
+
+const Item = props => {
+    const {
+        product_name,
+        product_sale_price,
+        quantity_ordered,
+        selected_options,
+        thumbnail,
+        url_key,
+        url_suffix
+    } = props;
+    const thumbnailUrl = thumbnail ? thumbnail.url : '';
```

Comment:
This certainly solves the issue when `thumbnail` is `undefined` but did you figure out why it was undefined and how we ended up here at all? I'd rather fix that than the side effect :)

Reply:
I presume that is because parts of the data is mock data which is resolved immediately in one query and the thumbnail is part of another query which is taking long.

If you look at how the `items` component renders the `item` component, you would see that I am spreading the `item` object and the `mappedImageData` object. `item` object is readily available while the `mappedImageData` takes a while to be resolved. `thumbnail` is part the `mappedImageData` hence it is undefined at first but on subsequent renders it is available.

Reply:
So what happens the first time we render before the image data query resolves? Does the UI attempt to render an invalid image using `url=""` or something? Can you check that out? Do we skip render of the thumbnail until we have it?

Reply:
If `url=''` no image will be rendered also no error will be thrown.

As I was writing this comment, I realized it is semantically not correct to render an image tag with an empty source so I will make the change to not render the component till the src is available. I can use the loading state to show a loading indicator till the src is available.

Reply:
I have pushed d03d1a2 which will not render the order details if the data is loading. Let me know what you think.

Reply:
Ok - now you don't need this line, and can just reference `thumbnail` directly. I'd also make the prop type required.

```suggestion
```

Reply:
Sorry forgot to remove that line of code.